### PR TITLE
feat: sort breadcrumbs with simple flow data

### PR DESCRIPTION
**Pull request by builditben**
Previously, the `sortBreadcrumbs` function required flow data to be normalized beforehand. To improve performance and make the interface easier to use, this function now only requires simple flow data (i.e. stored json data for a flow).

This refactor improves performance significantly and also drops unused fields from the output.

### CODE REVIEW

File: src/logic.test.ts [hunk 1]
```diff
@@ -2,27 +2,28 @@ import { sortFlow, normalizeFlow, sortBreadcrumbs } from "./logic";
 import * as sectioned from "./mocks/section-flow-breadcrumbs";
 import * as simple from "./mocks/simple-flow-breadcrumbs";
 import * as complex from "./mocks/complex-flow-breadcrumbs";
+import * as large from "./mocks/large-real-life-flow";
 import type { OrderedFlow, NormalizedFlow, OrderedBreadcrumbs } from "./types";
 
 describe("sortFlow", () => {
   test("it sorts a simple graph of nodes into an ordered array", () => {
     const sortedFlowNodes: OrderedFlow = sortFlow(simple.flow);
-    expect(sortedFlowNodes.length).toBe(6);
```
Comment (builditben) [inline src/logic.test.ts:10]: length checks are implicit in the equality check below

File: src/logic.test.ts [hunk 2]
```diff
@@ -2,27 +2,28 @@ import { sortFlow, normalizeFlow, sortBreadcrumbs } from "./logic";
...
     expect(sortedFlowNodes).toEqual(sectioned.orderedFlow);
   });
 
   test("it sorts a complex graph of nodes into an ordered array", () => {
     const sortedFlowNodes: OrderedFlow = sortFlow(complex.flow);
-    expect(sortedFlowNodes.length).toBe(27);
     expect(sortedFlowNodes).toEqual(complex.orderedFlow);
   });
 
+  test("it sorts a very large (5MB) graph of nodes into an ordered array within 2 seconds", () =>
```
Comment (DafyddLlyr) [inline src/logic.test.ts:24]: Really concrete and measurable test cases 👍

File: src/logic.ts [hunk 3]
```diff
@@ -33,53 +32,65 @@ export function sortFlow(flow: FlowGraph): OrderedFlow {
...
+        answers: foundCrumb.answers,
+        data: foundCrumb.data,
+        override: foundCrumb.override,
+        feedback: foundCrumb.feedback,
+      });
+    }
+    foundNode.edges?.forEach((childEdgeId) => {
+      searchNodeEdges(childEdgeId);
+    });
+    // short-curcuit if complete
```
Comment (DafyddLlyr) [inline src/logic.ts:87]:
```suggestion
    // short-circuit if complete
```

**Review by DafyddLlyr — APPROVED** [review]
Looks good 👍 

One very very minor nit that might help with keeping tests up to date and meaningful - can we add a comment to the begging of each mock file to give a little context?

```ts
// Service: "apply-for-lawful-development-certificate", Buckinghamshire
// Date: XX/XX/23
// ID: abc123
```

**Review by jessicamcinchak — APPROVED** [review]

**Pull request merged**